File: docs-examples/example-java/src/test/java/io/micronaut/configuration/rabbitmq/docs/consumer/connection/ProductClient.java
package io.micronaut.configuration.rabbitmq.docs.consumer.connection;

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Binding;
import io.micronaut.configuration.rabbitmq.annotation.RabbitClient;
import io.micronaut.context.annotation.Requires;
// end::imports[]

@Requires(property = "spec.name", value = "ConnectionSpec")
// tag::clazz[]
@RabbitClient // <1>
public interface ProductClient {

    @Binding(value = "product", connection = "product-cluster") // <2>
    void send(byte[] data); // <3>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/consumer/connection/ProductClient.kt
package io.micronaut.rabbitmq.docs.consumer.connection

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.rabbitmq.annotation.Binding
import io.micronaut.rabbitmq.annotation.RabbitClient
// end::imports[]

@Requires(property = "spec.name", value = "ConnectionSpec")
// tag::clazz[]
@RabbitClient // <1>
interface ProductClient {

    @Binding(value = "product", connection = "product-cluster") // <2>
    fun send(data: ByteArray)  // <3>
}
// end::clazz[]
